### PR TITLE
Reorganize accrual code and fix 6-month cutoff

### PR DESCRIPTION
**Pull request by cyrossignol**
This pulls some changes from the branch for the upcoming superblock window accrual updates to fix an issue seen on testnet that allows a research reward claim after the 6-month cutoff due to a missed condition in #1583. I added a testnet exception for this block as discussed because we don't want to roll-back testnet in this case.

The rest of the commits in this PR are mostly cosmetic changes that move around tally code to better organize accrual computer classes.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -2588,8 +2514,8 @@ bool CBlock::ConnectBlock(CTxDB& txdb, CBlockIndex* pindex, bool fJustCheck, boo
                 vtx[1].GetHash().ToString().substr(0,10));
         }
 
-        //9-3-2015
-        int64_t nMaxResearchAgeReward = NN::ResearchAgeComputer::MaxReward(nTime);
+        // TODO: determine max reward from accrual computer implementation:
+        int64_t nMaxResearchAgeReward = 12750 * COIN;
```
Comment (jamescowens) [inline src/main.cpp:2518]: Why did you revert this to a hardcoded constant?

**Review by jamescowens — APPROVED** [review]
I am curious about the hardcoded 12750 directly in main.cpp

File: src/main.cpp (hunk 1 above)
Comment (cyrossignol) [reply src/main.cpp:2518]: The `ResearchAgeComputer` type is now internal to the implementation of the tally interface, so `MaxReward()` isn't exposed directly. I didn't want to pull in the more complex changes from this branch just for the fix to compile, so I put in the hard-coded constant as a stand-in until the rest of the superblock window changes are finished.

Comment (jamescowens) [reply src/main.cpp:2518]: Makes sense.

**Pull request merged**